api/app/Jobs/SendNotificationBeforeEvent.php

```diff
@@ -43,16 +48,26 @@ public function handle(EventRepository $eventRepository, ZoomService $zoomServic
         $events = $eventRepository->findByCriteria(...$criteria);
 
         foreach ($events as $event) {
-            if ($event->eventType->location_type == 'zoom') {
-                $event->zoom_meeting_link = $zoomService->meeting($event);
-                $event->save();
-            }
-
+            $this->saveMeetingLink($event->eventType->location_type, $event, $zoomService, $whaleService);
             $event->eventType->owner->notify(new NotificationBeforeEventForOwner($event));
             Notification::route('mail', $event->invitee_email)
                 ->notify(new NotificationBeforeEventForInvitee($event));
             $event->notified = true;
             $eventRepository->save($event);
         }
     }
+
+    private function saveMeetingLink(string $location, Event $event, ZoomService $zoomService, WhaleService $whaleService): void
+    {
+        switch ($location) {
+            case $location == 'zoom':
+                $event->zoom_meeting_link = $zoomService->meeting($event);
+                $event->save();
+            break;
+            case $location == 'whale':
+                $event->whale_meeting_link = $whaleService->meeting($event);
+                $event->save();
+            break;
+        }
+    }
```

Comment:
This construction is wrong
https://www.php.net/manual/ru/control-structures.switch.php